src/i18n.test.ts

```diff
@@ -0,0 +1,210 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+
+// Mock the JSON imports before importing i18n module
+vi.mock('./locales/en/main.json', () => ({ default: { welcome: 'Welcome' } }))
+vi.mock('./locales/en/nodeDefs.json', () => ({
+  default: { testNode: 'Test Node' }
+}))
+vi.mock('./locales/en/commands.json', () => ({
+  default: { save: 'Save' }
+}))
+vi.mock('./locales/en/settings.json', () => ({
+  default: { theme: 'Theme' }
+}))
+
+// Mock lazy-loaded locales
+vi.mock('./locales/zh/main.json', () => ({ default: { welcome: '欢迎' } }))
+vi.mock('./locales/zh/nodeDefs.json', () => ({
+  default: { testNode: '测试节点' }
+}))
+vi.mock('./locales/zh/commands.json', () => ({ default: { save: '保存' } }))
+vi.mock('./locales/zh/settings.json', () => ({ default: { theme: '主题' } }))
+
+describe('i18n', () => {
+  beforeEach(async () => {
+    vi.resetModules()
+  })
+
+  describe('mergeCustomNodesI18n', () => {
+    it('should immediately merge data for already loaded locales (en)', async () => {
+      const { i18n, mergeCustomNodesI18n } = await import('./i18n')
```

Comment:
Can you import these at the top with the normal import syntax?

Reply:
fixed